### ISSUE TITLE
Non-deterministic module ordering and chunk hashes since v0.11.13

### ISSUE THREAD

**Issue by eelco** [evanw/esbuild#1304]
Observed when building our code base twice:

Run 1:
```
import {
  Folder,
  fileManager,
  updateModelErrors
} from "https://app.framerstatic.com/chunk-7MI3WOZG.js";
import "https://app.framerstatic.com/chunk-KE3Z4VU4.js";
import "https://app.framerstatic.com/chunk-V6WDQTIM.js";
import "https://app.framerstatic.com/chunk-BBCUBSXY.js";
import {
  EditableFile,
  dirname,
  log,
  nameAndExtensionFor,
  pathjoin,
  restoreExtension
} from "https://app.framerstatic.com/chunk-2LPVDXEF.js";
import "https://app.framerstatic.com/chunk-VHU2THJN.js";
import "https://app.framerstatic.com/chunk-OK5YLNGZ.js";
```

Run 2:
```
import {
  Folder,
  fileManager,
  updateModelErrors
} from "https://app.framerstatic.com/chunk-ZYEJSYSI.js";
import "https://app.framerstatic.com/chunk-OK5YLNGZ.js";
import {
  EditableFile,
  dirname,
  log,
  nameAndExtensionFor,
  pathjoin,
  restoreExtension
} from "https://app.framerstatic.com/chunk-7GKJRT75.js";
import "https://app.framerstatic.com/chunk-BBCUBSXY.js";
import "https://app.framerstatic.com/chunk-V6WDQTIM.js";
import "https://app.framerstatic.com/chunk-KE3Z4VU4.js";
import "https://app.framerstatic.com/chunk-VHU2THJN.js";
```

Most of the chunk hashes appear in both files, but in a completely different order. This obviously affects the chunk hashes of the files (as demonstrated by the first import).

The issue first occurs in 0.11.13 (and also the latest version, so I assume all subsequent versions, I’ve also tested with 0.11.12, 0.11.14, 0.11.17, 0.11.23 and 0.12.1). 

This might be the return of https://github.com/evanw/esbuild/issues/1075, but looking at the [comparison](https://github.com/evanw/esbuild/compare/v0.11.12...v0.11.13) I suspect https://github.com/evanw/esbuild/commit/65a010fd2ed1ea5ff88b380196f246cbc638dcfd might be the culprit.

**Comment by evanw** [comment]
Thanks for the report. I agree with your assessment. The next release will contain a fix that should hopefully solve this regression.

**Comment by evanw** [comment]
The fix has been published as version 0.12.4. Can you try it out and see if that fixes this problem?

**Comment by eelco** [comment]
Yes, that seems to fix it, thank you!
